src/main/java/pl/javastart/task/internal/Mix.java

```diff
@@ -0,0 +1,72 @@
+package pl.javastart.task.internal;
+
+public class Mix extends Prepaid {
+
+    int smsLimit;
+    int mmsLimit;
+    int secLimit;
+
+    public Mix(double accountState, int smsLimit, int mmsLimit, int secLimit, double smsPrice, double mmsPrice, double perMinPrice) {
+        super(accountState, smsPrice, mmsPrice, perMinPrice);
+        this.smsLimit = smsLimit;
+        this.mmsLimit = mmsLimit;
+        this.secLimit = secLimit;
+    }
+
+    @Override
+    public void sendSms() {
+        if (smsLimit > 0) {
+            System.out.println("SMS wysłany\n");
+            smsLimit--;
+            smsCount++;
+        } else {
+            super.sendSms();
+        }
+    }
+
+    @Override
+    public void sendMms() {
+        if (mmsLimit > 0) {
+            System.out.println("MMS wysłany\n");
+            mmsLimit--;
+            mmsCount++;
+        } else {
+            super.sendMms();
+        }
+    }
+
+    @Override
+    public void makePhoneCall(int seconds) {
+
+        if (secLimit <= 0 && accountState < (pricePerMin / 60)) {
+            System.out.println("Brak środków na koncie\n");
+            return;
+        }
+        int secondsFromLimit = 0;
+        if (secLimit > 0) {
+            secondsFromLimit = useSecondsFromLimit(seconds);
+        }
+        int secondsFromAccount = 0;
+        if (accountState > 0 && secondsFromLimit < seconds) {
+            secondsFromAccount = super.useSecondsFromAccount(seconds - secondsFromLimit);
+        }
+        int callDuration = secondsFromAccount + secondsFromLimit;
+        secCount += callDuration;
+        System.out.printf("Rozmowa trwała %d sekund\n\n", callDuration);
+    }
```

Comment:
Może zostać tak, jak jest, ale tutaj warto by było wykorzystać logikę z klasy nadrzędnej, tzn.:
- jeżeli są darmowe sekundy to ich użyj,
- jeżeli rozmowa wciąż trwa, to wywołaj metodę z klasy Prepaid i spróbuj użyć środków na koncie (`super.makePhoneCall(remainingCallSeconds)`)
U Ciebie jest to o tyle kłopotliwe, że w klasach reprezentujących umowy masz System.out.printlny (w rzeczywistości miejscem interakcji z użytkownikiem powinien być raczej telefon, a nie umowa/karta sim)